Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -158,14 +158,25 @@ void BTDiagnostics::DerivedInitData ()
     // j >= i / gamma / (1+beta) * dt_snapshot / dt_boosted_frame
     const int final_snapshot_starting_step = static_cast<int>(std::ceil(final_snapshot_iteration / warpx.gamma_boost / (1._rt+warpx.beta_boost) * m_dt_snapshots_lab / dt_boosted_frame));
     const int final_snapshot_fill_iteration = final_snapshot_starting_step + num_buffers * m_buffer_size - 1;
+
     if (final_snapshot_fill_iteration > warpx.maxStep()) {
-        std::string warn_string =
-            "\nSimulation might not run long enough to fill all BTD snapshots.\n"
-            "Final step: " + std::to_string(warpx.maxStep()) + "\n"
-            "Last BTD snapshot fills around step: " + std::to_string(final_snapshot_fill_iteration);
-        ablastr::warn_manager::WMRecordWarning(
-            "BTD", warn_string,
-            ablastr::warn_manager::WarnPriority::low);
+        if (warpx.compute_max_step_from_btd) {
+            warpx.updateMaxStep(final_snapshot_fill_iteration);
+            amrex::Print()<<"max_step insufficient to fill all BTD snapshots. Automatically increased to: "
+                <<final_snapshot_fill_iteration<<std::endl;
+        } else {
+            std::string warn_string =
+                "\nSimulation might not run long enough to fill all BTD snapshots.\n"
+                "Final step: " + std::to_string(warpx.maxStep()) + "\n"
+                "Last BTD snapshot fills around step: " + std::to_string(final_snapshot_fill_iteration);
+            ablastr::warn_manager::WMRecordWarning(
+                "BTD", warn_string,
+                ablastr::warn_manager::WarnPriority::low);
+        }
```

Comment:
Do I understand the code behavior correctly?
- If both `max_step` and `compute_max_step_from_btd` are set in the input, and if `max_step` is not enough to fill all BTD, then we overwrite `max_step` and print a message;
- If `max_step` is set in the input but `compute_max_step_from_btd`  isn't, and if `max_step` is not enough to fill all BTD, then we print a warning, but do not overwrite `max_step` (so we end up not filling all BTD);
- If `max_step` is not set in the input but `compute_max_step_from_btd` is, then `max_step` is set automatically.

Question: shouldn't we want to kind of merge points 1 and 2 above, namely overwrite `max_step` regardless of whether `compute_max_step_from_btd` is set in the input? In this case, the new parameter `compute_max_step_from_btd` would be still useful for users who do not want to specify `max_step`.

Happy to discuss more offline, if my comment isn't clear.

Reply:
Yes, the PR was specifically designed to achieve these 3 points.  I'm glad they came out so clearly.  I thought about combining points 1 and 2, but I didn't want to overwrite ``max_step`` unless a user specifically requests for this to be done.  The decision to combine points 1 and 2 seems beyond my paygrade to make, I yield to higher authorities